### DISCUSSION TITLE
Match icons behavior with Notifications by allowing them to show up only when needed

### DISCUSSION THREAD

**Discussion by KodeXall** [jonaskohl/CapsLockIndicator#125] [Ideas]
Right now a user might choose to see a permanent notification for a state when a key is turned ON and a notification lasting for up to 2 seconds for when a key is turned OFF. But for icons it is impossible - i.e. the user needs to choose to either always see icons or to not see them at all; plus a user is able to move icons between hidden icons Tray sub-menu and that "main" Tray part visible on Taskbar [thus visible always - assuming the Taskbar is locked]

In my opinion a user should also be able to choose to see an icon only for the ON [or OFF] state in form of an icon on the Tray's Taskbar area. Tray is a valuable space and although its space is theoretically not limited, it becomes easily cluttered thus hardly informative

**Comment by jonaskohl** [comment]
I don't know if disappearing and reappearing tray icons are particularly good UX because of all other icons shifting around, but if this is a feature more users could benefit from, maybe I'll consider adding it at least to the advaced options

**Reply by KodeXall** [reply]
Yeas, I also have to admit that icons icons in Tray are shifting around because of various programs, which is counterproductive. I tried arranging them but it just does not work, as something like Media Player Classic can put in it 3 and more icons and then I can also have a second instance of it, plus some rarely used software that I did not arrange in Tray prior to this will add its icon etc.
